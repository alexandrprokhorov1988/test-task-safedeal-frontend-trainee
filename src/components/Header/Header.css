.header {
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 95%;
  max-width: 728px;
}

.header__title {
  padding: 0;
  margin: 0;
  color: #000000;
  text-transform: uppercase;
  font-size: 36px;
  line-height: 42px;
  padding-top: 15px;
}
