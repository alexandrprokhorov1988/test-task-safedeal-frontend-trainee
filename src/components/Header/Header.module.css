.header {
  margin: 0 auto;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 95%;
  max-width: 728px;
}

.title {
  padding: 0;
  margin: 0;
  color: #000000;
  text-transform: uppercase;
  font-size: 36px;
  line-height: 42px;
  padding-top: 15px;
  text-shadow: 0 2.8px 2.2px rgba(0, 0, 0, 0.034),
  0 6.7px 5.3px rgba(0, 0, 0, 0.048),
  0 12.5px 10px rgba(0, 0, 0, 0.06),
  0 22.3px 17.9px rgba(0, 0, 0, 0.072),
  0 41.8px 33.4px rgba(0, 0, 0, 0.086),
  0 100px 80px rgba(0, 0, 0, 0.12);
}
