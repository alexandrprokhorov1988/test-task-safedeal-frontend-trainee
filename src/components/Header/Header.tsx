import React from 'react';
import style from './Header.module.css';

/**
 * Header component.
 */
const Header: React.FC = () => {
  console.log('header');  // todo del
  return (
    <footer className={style.header}>
      <h1 className={style.title}>Test APP</h1>
    </footer>
  );
};

export default Header;
