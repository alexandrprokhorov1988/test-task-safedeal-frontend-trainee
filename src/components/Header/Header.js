import React from 'react';
import './Header.css';

function Header() {

  return (
    <footer className="header">
      <h1 className="header__title">Test APP</h1>
    </footer>
  );
}

export default Header;
