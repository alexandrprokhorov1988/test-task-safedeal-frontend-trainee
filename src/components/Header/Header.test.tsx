import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import Header from './Header';

let container:any = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Header component', () => {
  it("Рендер Header с текстом Test APP", () => {
    act(() => {
      render(<Header/>, container);
    });
    expect(container.textContent).toBe("Test APP");
  });

  it('Рендер Header snapshot', () => {
    const tree = renderer
      .create(<Header/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
