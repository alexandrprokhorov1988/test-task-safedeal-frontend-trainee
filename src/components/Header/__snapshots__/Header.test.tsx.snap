// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Рендер Header snapshot 1`] = `
<footer
  className="header"
>
  <h1
    className="title"
  >
    Test APP
  </h1>
</footer>
`;
